subsys/fs/nvs/nvs.c

```diff
@@ -444,7 +536,11 @@ static int nvs_recover_last_ate(struct nvs_fs *fs, uint32_t *addr)
 
 	ate_size = nvs_al_size(fs, sizeof(struct nvs_ate));
 
-	*addr -= ate_size;
+	if (NVS_DEVICE_HAS_NO_ERASE(fs)) {
+		*addr -= 2 * ate_size;
```

Comment:
I think that it would nice to provide a description here, why the two ATEs are needed here; I know what they are for and why that differs from original design but it may not be clear from the code at the first glance.